### PR TITLE
Filter out unused json_path tokens

### PR DESCRIPTION
**Pull request by revans2**
This is related to https://github.com/NVIDIA/spark-rapids-jni/pull/2047

The code in spark-rapids-jni filters out the path tokens that it does not care about. But that is not clear how it is happening. It also makes the error message confusing because `$.foo` has a depth of 2, even though most would expect it to have a depth of 1.  This does the filtering on the plugin side and updates the message about falling back to the CPU to be more user friendly. https://github.com/NVIDIA/spark-rapids-jni/pull/2047 will be updated to remove the unused/unneeded token enums entirely.

### CODE REVIEW

**Comment by revans2** [comment]
build

File: sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala [hunk 1]
```diff
@@ -150,8 +156,11 @@ class GpuGetJsonObjectMeta(
     lit.foreach { l =>
       val instructions = JsonPathParser.parse(l.value.asInstanceOf[UTF8String].toString)
       if (!conf.isLegacyGetJsonObjectEnabled) {
-        if (instructions.exists(JsonPathParser.fallbackCheck)) {
-          willNotWorkOnGpu("get_json_object on GPU does not support more than 16 nested paths")
+        val updated = instructions.map(JsonPathParser.filterInstructionsForJni)
+        if (updated.exists(JsonPathParser.fallbackCheck)) {
+          willNotWorkOnGpu(s"get_json_object on GPU does not support more " +
+            s"than ${JSONUtils.MAX_PATH_DEPTH} nested paths." +
+            s"(Found ${instructions.map(_.length)})")
```
Comment (jlowe) [inline sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala:163]: Nit: I think this isn't going to display nicely, will probably have Seq( or something on it that isn't user friendly.  Could use .mkString to tidy this up or just report the max.

**Review by jlowe — APPROVED** [review]
Small formatting nit, lgtm.

**Review by res-life — DISMISSED** [review]
LGTM

**Comment by revans2** [comment]
build

**Comment by revans2** [comment]
I just merged in https://github.com/NVIDIA/spark-rapids-jni/pull/2047

The hang in the tests is being caused by the mismatch between the tests. I will try and trigger the builds to hopefully get the dependency in ASAP

**Comment by revans2** [comment]
@jlowe please take another look

**Review by jlowe — APPROVED** [review]

**Comment by sameerz** [comment]
build

**Pull request merged**